shims/spark311db/src/main/scala/com/nvidia/spark/rapids/shims/spark311db/ParquetCachedBatchSerializer.scala

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims.spark311db
+
+import com.nvidia.spark.rapids.shims
+
+class ParquetCachedBatchSerializer extends shims.spark311.ParquetCachedBatchSerializer {
```

Comment:
this file shouldn't be needed now right?  the 311 shim version of this works.
Or is the intention to keep the db one so that the user specifies this one? https://nvidia.github.io/spark-rapids/docs/additional-functionality/cache-serializer.html.   If that is the case we need to update the docs.

originally the intention was that since the user has to specify it have it match the version of spark they are using so that its hopefully least confusing.    I don't know how much it betters if they specify spark311 vs spark311db

Reply:
if its possible it would be nice to have one class that just loaded the proper shim version, but that is a separate issue.

Reply:
> this file shouldn't be needed now right? the 311 shim version of this works.
> Or is the intention to keep the db one so that the user specifies this one? https://nvidia.github.io/spark-rapids/docs/additional-functionality/cache-serializer.html. If that is the case we need to update the docs.
> 
> originally the intention was that since the user has to specify it have it match the version of spark they are using so that its hopefully least confusing. I don't know how much it betters if they specify spark311 vs spark311db

If we add spark311db to the documentation we will then also have to add spark311cdh. I almost feel like they are all spark 311 and we aren't doing anything specific in the extended version of their PCBS we should just get rid of them. 

> if its possible it would be nice to have one class that just loaded the proper shim version, but that is a separate issue.

This is a good idea, I can look into it as a follow-on

Reply:
@jlowe thoughts?

Reply:
> if its possible it would be nice to have one class that just loaded the proper shim version, but that is a separate issue.

☝️ This.  We should not have Spark-specific versions of user-visible classes unless they are truly required (e.g.: as in the shuffle case, unfortunately).  If we know one class will work going forward, as is the case with the main executor plugin, then we should strive to use a common class name without a Spark version in it.  If this is indeed possible, we should deprecate the old 311 version and eventually remove it.

So it really all comes down to that question.  If we can have a common version, my vote is to use the one class.  We can change the package name and deprecate the existing spark311 package version in a new PR if it's too tricky to do in this one.

Reply:
OK. I can do this as a follow-on. In the interim, do we just update the doc or remove the spark311db or spark311cdh versions of the serializer? I feel removing the db and cdh versions of the serializer should be the way as we will do more work as part of the follow-on

Reply:
If we're planning on removing these classes in the near future then we should not document them only to rip them out immediately afterward.  Let's keep the number of classes to deprecate to a minimum.